src/hyp3_isce2/insar_tops_burst.py

```diff
@@ -211,34 +254,38 @@ def main():
         '--looks', choices=['20x4', '10x2', '5x1'], default='20x4', help='Number of looks to take in range and azimuth'
     )
     parser.add_argument(
-        '--apply-water-mask',
-        type=string_is_true,
-        default=False,
-        help='Apply a water body mask before unwrapping.',
-    )
-    parser.add_argument(
-        'granules',
-        type=str.split,
-        nargs='+',
-        help='Reference and secondary scene names'
+        '--apply-water-mask', type=string_is_true, default=False, help='Apply a water body mask before unwrapping.'
     )
+    parser.add_argument('--reference', type=str.split, nargs='+', help='List of reference scenes"')
+    parser.add_argument('--secondary', type=str.split, nargs='+', help='List of secondary scenes"')
 
     args = parser.parse_args()
 
-    granules = [item for sublist in args.granules for item in sublist]
-    if len(granules) != 2:
-        parser.error('No more than two granules may be provided.')
-    if len(granules[0]) != len(granules[1]) != 1:
-        parser.error('Must include 1 reference and 1 secondary.')
+    references = [item for sublist in args.reference for item in sublist]
+    secondaries = [item for sublist in args.secondary for item in sublist]
+    if len(references) < 1 or len(secondaries) < 1:
+        parser.error('Must include at least 1 reference and 1 secondary')
+    if len(references) != len(secondaries):
+        parser.error('Must have the same number of references and secondaries')
```

Comment:
Granule name validation is now spread across several places: the checks shown here, the check in `insar_tops_multi_burst` to see if the burst IDs match, the two calls to `check_older_granule_is_reference`, and the call to `validate_bursts` in `insar_tops_single_burst`. It would be great to combine these checks as much as possible (probably call them all at this location, right after parsing the CLI args), except for any checks that are specific to single or multi burst.

Reply:
Agree. I'm working on a fix for this. I'll let you know when it's done.

Reply:
OK, the solution I've come up with is moving all validation into the `burst.validate_bursts` function, then calling it inside of `insar_tops_single_burst` and `insar_tops_multi_burst`. I prefer calling it in these two places instead of `main` to ensure that users are getting validation when running the core functions from python.